### ISSUE TITLE
Structuring docs folder

### ISSUE THREAD

**Issue by andrzej-k** [intelsdi-x/snap#1390]
Do we need at this point to structure our ```docs``` folder? 
For example by introducing **developer** and **user** spaces:

**docs/developer**
BUILD_AND_TEST.md
DISTRIBUTED_WORKFLOW_ARCHITECTURE.md
PLUGIN_AUTHORING.md
PLUGIN_PACKAGING.md
PLUGIN_SIGNING.md
PROFILING.md

**docs/user**
METRICS.md
PLUGIN_CATALOG.md
PLUGIN_LIFECYCLE.md
REST_API.md
SNAPTEL.md
SNAPTELD.md
SNAPTELD_CONFIGURATION.md
TASKS.md
TRIBE.md
plugins.yml

**docs/**
AUTHORS.md
GLOSSARY.md
MAINTAINERS.md
(new) README.md - with guidance on what's in each space

**Comment by mjbrender** [comment]
TL;DR I strongly prefer this folder remains simple and flat where you can quickly ⌘+F and search for the right information.

I always like the goal of clarifying documentation, but I do want to avoid having a hard (folder) boundary between developer and user. We ran into this issue in Riak documentation and had a massive amount of user feedback that the boundaries don't make sense. I like where they landed in the [latest version of docs](http://docs.basho.com/riak/kv/2.2.0/), but we're honestly not there yet.

When we are there, we should think beyond the `docs/` folder and look at a proper tool like [Read The Docs](https://readthedocs.org). Would you be supportive of going that direction @andrzej-k?

**Comment by andrzej-k** [comment]
@mjbrender Thank you for sharing your view. I think you presented valid points - I'd definitely support the direction you propose. 
I'm going to close this issue - I believe any further doc updates deserve separate discussion.
